extensions/ql-vscode/src/query-history.ts

```diff
@@ -680,6 +680,10 @@ export class QueryHistoryManager extends DisposableObject {
     return this.treeDataProvider.getCurrent();
   }
 
+  getQueryById(queryId: string): RemoteQueryHistoryItem | undefined {
+    return this.treeDataProvider.allHistory.find(i => i.t === 'remote' && i.queryId === queryId) as RemoteQueryHistoryItem;
```

Comment:
Why are we filtering only for remote queries? Is it to return `RemoteQueryHistoryItem`? In that case perhaps the function should be called `getRemoteQueryById`?

It'd be nice if we just had a generic `getQueryById` that returns QueryHistoryItem's though rather than anything more specific.

Reply:
I'll rename it to `getRemoteQueryById`. The reason for not returning any local queries is that local queries [don't have a query ID](https://github.com/github/vscode-codeql/blob/1b6685ef6ff208812af7350be013c842dcefd484/extensions/ql-vscode/src/query-results.ts#L213), so the `i.t === 'remote'` is necessary to even be able to access the `queryId` field. Without it, the union type doesn't necessarily have a `queryId` field.